### ISSUE TITLE
[Module Proposal]: `avm/ptn/lz/sub-vending`

### ISSUE THREAD

**Issue by sebassem** [Azure/Azure-Verified-Modules#548]
### Check for previous/existing GitHub issues/module proposals

- [X] I have checked for previous/existing GitHub issues/module proposals.

### Check this module doesn't already exist in the module indexes

- [X] I have checked for that this module doesn't already exist in the module indexes; or I'm proposing the module to be migrated from CARML/TFVM.

### Bicep or Terraform?

Bicep

### Module Classification?

Pattern Module

### Module Name

avm/ptn/lz/sub-vending

### Module Details

A pattern module for the [landing zone subscription vending module.](https://github.com/Azure/bicep-lz-vending). This module is already using AVM resource modules in most resources being deployed except for a couple that are not yet published.

### Do you want to be the owner of this module?

Yes

### Module Owner's GitHub Username (handle)

jtracey93

### (Optional) Secondary Module Owner's GitHub Username (handle)

sebassem

**Comment by matt-FFFFFF** [comment]
Hi @sebassem 

This exists today outside AVM and is owned by myself. We do not want to create an AVM equivalent until we have worked out the migration path for the as-is.

https://github.com/Azure/terraform-azurerm-lz-vending/

**Comment by jtracey93** [comment]
Sorry team, should of gave a heads up (will do on todays call).

But this is part of work that I am aware of and am asking @sebassem to do to migrate `bicep-lz-vending` to AVM 👍

cc: @matt-FFFFFF @matebarabas

**Comment by matebarabas** [comment]
Thanks for clarifying @jtracey93! @sebassem, before doing the "handshake" and officially capturing this one, we'd need to work a bit on the name of this module to make it compatible with the naming conventions. Something like this would work: `avm/ptn/alz/sub-vending` 
Would this work? Thanks! cc: @jtracey93

**Comment by sebassem** [comment]
> Thanks for clarifying @jtracey93! @sebassem, before doing the "handshake" and officially capturing this one, we'd need to work a bit on the name of this module to make it compatible with the naming conventions. Something like this would work: `avm/ptn/alz/sub-vending` Would this work? Thanks! cc: @jtracey93

Sounds good to me

**Comment by matebarabas** [comment]
Hi @sebassem,

Thanks for requesting/proposing to be an AVM module owner!

We just want to confirm **you agree to the below pages** that define what module ownership means:

- [Team Definitions & RACI](https://azure.github.io/Azure-Verified-Modules/specs/shared/team-definitions)
- [Shared Specification (Bicep & Terraform)](https://azure.github.io/Azure-Verified-Modules/specs/shared)
- [Module Support](https://azure.github.io/Azure-Verified-Modules/help-support/module-support)

Any questions or clarifications needed, let us know!

If you agree, please just **reply to this issue with the exact sentence below** (as this helps with our automation 👍):

"I CONFIRM I WISH TO OWN THIS AVM MODULE AND UNDERSTAND THE REQUIREMENTS AND DEFINITION OF A MODULE OWNER"

Thanks,

The AVM Core Team

#RR

**Comment by matebarabas** [comment]
After internally aligning with @jtracey93, I'm changing the name to 'avm/ptn/lz/sub-vending'. Jack also asked me to put him down as the primary owner, so I'll capture you @sebassem as the secondary owner.

**Comment by matebarabas** [comment]
Hi @jtracey93,

Thanks for requesting/proposing to be an AVM module owner!

We just want to confirm **you agree to the below pages** that define what module ownership means:

- [Team Definitions & RACI](https://azure.github.io/Azure-Verified-Modules/specs/shared/team-definitions)
- [Shared Specification (Bicep & Terraform)](https://azure.github.io/Azure-Verified-Modules/specs/shared)
- [Module Support](https://azure.github.io/Azure-Verified-Modules/help-support/module-support)

Any questions or clarifications needed, let us know!

If you agree, please just **reply to this issue with the exact sentence below** (as this helps with our automation 👍):

"I CONFIRM I WISH TO OWN THIS AVM MODULE AND UNDERSTAND THE REQUIREMENTS AND DEFINITION OF A MODULE OWNER"

Thanks,

The AVM Core Team

#RR

**Comment by matebarabas** [comment]
Hi @jtracey93,

Thanks for confirming that you wish to own this AVM module and understand the related requirements and responsibilities!

Before starting development, please ensure ALL the following requirements are met.

**Please use the following values explicitly as provided in the [module index](https://azure.github.io/Azure-Verified-Modules/indexes/) page**:

- For your module:
  - `ModuleName` - for naming your module
  - `TelemetryIdPrefix` - for your module's [telemetry](https://azure.github.io/Azure-Verified-Modules/specs/shared/#id-sfr3---category-telemetry---deploymentusage-telemetry)
- For your module's repository:
  - Repo name and folder path are defined in `RepoURL`
  - Create GitHub teams for module owners and contributors and grant them permissions as outlined [here](https://azure.github.io/Azure-Verified-Modules/specs/shared/#id-snfr20---category-contributionsupport---github-teams-only).
  - Grant permissions for the AVM core team and PG teams on your GitHub repo as described [here](https://azure.github.io/Azure-Verified-Modules/specs/shared/#id-snfr9---category-contributionsupport---avm--pg-teams-github-repo-permissions).

Check if this module exists in the other IaC language. If so, collaborate with the other owner for consistency. 👍

You can now start the development of this module! ✅ Happy coding! 🎉

**Please respond to this comment and request a review from the AVM core team once your module is ready to be published! Please include a link pointing to your PR, once available. 🙏**

Any further questions or clarifications needed, let us know!

Thanks,

The AVM Core Team

**Comment by sebassem** [comment]
we are currently planning how the pattern module should like look. Work is in progress

**Comment by matebarabas** [comment]
@sebassem, do you have any updates on this issue? Thanks!

CC: @jtracey93

**Comment by sebassem** [comment]
> @sebassem, do you have any updates on this issue? Thanks!
> 
> CC: @jtracey93

Currently working on it. There are still some pieces that needs to be implemented on the CI piece so working with Alexander to get those in so we can continue the work on the module

**Comment by sebassem** [comment]
Still work in progress

**Comment by matebarabas** [comment]
Is there a draft PR already? (If there isn't, do you already have an ETA?))

**Comment by sebassem** [comment]
> Is there a draft PR already? (If there isn't, do you already have an ETA?))

We are working with Alexander on some changes that need to take place in AVM scripts to accommodate this and future pattern modules, some of those changes are already done and we are still working with him on the rest. Once they are there, we would be ready to submit the PR hopefully

**Comment by sebassem** [comment]
Draft PR: https://github.com/Azure/bicep-registry-modules/pull/2080

**Comment by matebarabas** [comment]
- [x] Module published.
- [x] GitHub teams are available and assigned to their parents.
- [x] `CODEOWNERS` file updated.
 
Thank you for your work @sebassem! I'm closing this issue now.
